scalafmt-core/shared/src/main/scala/org/scalafmt/config/ProjectFiles.scala

```diff
@@ -70,7 +70,14 @@ object ProjectFiles {
     private def regex(seq: Seq[String]) = create(seq, new Regex(_))
 
     private final class Nio(pattern: String) extends PathMatcher {
-      private val matcher = fs.getPathMatcher(pattern)
+      private val matcher =
+        try { fs.getPathMatcher(pattern) }
+        catch {
+          case _: IllegalArgumentException =>
+            throw new ScalafmtConfigException(
```

Comment:
we should do the same in `Regex` below, and also catch `ReflectionException` in `ScalafmtReflectConfig.projectMatcherField`.

Reply:
Added with a new test case